SWENT-GRP5/SOLVIT
# Fix: Requests feed for Providers

### Description:
This pull request enhances the `ListRequestsFeedScreen`, fixing display issues and improving the overall user experience and code clarity. The search bar and service-type filter functionalities have also been added, enabling users to efficiently search and filter service requests by type.

- **What issue does it solve?** This PR resolves a bug which displayed the Service Request ID in the `ListRequestsFeedScreen` and significantly improves the screen’s appearance and usability, making it more intuitive for users to view and filter service requests.
- **What functionality does it add?** It refines the UI of `ListRequestsFeedScreen`, implements the logic behind the search bar for finding specific requests and the service-type filter. These additions enhance user experience. The code readability and structure have also been revised.

### Related Issues:
- #288

### Changes Made:
- Fixed Service Request ID display bug in `ListRequestsFeedScreen`.
- Modified UI and code structure for improved readability and user experience.
- Implemented the logic for the search bar and filter functionality to allow users to search and filter by service type.
- Updated `ListServicesRequestsScreenTest` to accommodate the new UI elements and logic.

### How to Test:
1. Navigate to `ListRequestsFeedScreen` and confirm the requests ID are no longer displayed.
2. Use the search bar to look up specific service requests and verify that it functions correctly.
3. Apply the service-type filter to view requests by type and confirm the filtering works as expected.
4. Run the updated tests in `ListServicesRequestsScreenTest` to validate functionality.

---

### Checklist:

--- 

Here are some screenshots to see what it looks like:
<img width="336" alt="Capture d’écran 2024-11-09 à 20 35 34" src="https://github.com/user-attachments/assets/ffcf921c-02f6-457c-97b1-676bf7172946">
<img width="336" alt="Capture d’écran 2024-11-09 à 20 35 47" src="https://github.com/user-attachments/assets/da8ce8dc-34d7-4904-ac64-43581148048d">
<img width="336" alt="Capture d’écran 2024-11-09 à 20 36 08" src="https://github.com/user-attachments/assets/088e2aa3-abdf-4eb6-ac92-6c87ddb437ec">
